### ISSUE TITLE
Installation fails on python 3.8

### ISSUE THREAD

**Issue by daniel-klein** [pydata/numexpr#352]
I have a fresh installation of python 3.8.1 and have a dependency on numexpr.  Unfortunately, the installation is failing with the attached log.  The final message is:

```console
2019-12-23T12:14:06,218 pip._internal.exceptions.InstallationError: Command errored out with exit status 1: /Users/dklein/venv/IDM/bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/t9/ljxsv_z97ln21qgj3k11r_2w0000gn/T/pip-install-k5mglaxw/numexpr/setup.py'"'"'; __file__='"'"'/private/var/folders/t9/ljxsv_z97ln21qgj3k11r_2w0000gn/T/pip-install-k5mglaxw/numexpr/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /private/var/folders/t9/ljxsv_z97ln21qgj3k11r_2w0000gn/T/pip-record-z9ps0lv7/install-record.txt --single-version-externally-managed --compile --install-headers /Users/dklein/venv/IDM/include/site/python3.8/numexpr Check the logs for full command output.
```

Host OS is macOS Catalina v10.15.1.  Thanks for any advice you may have.

[numexpr.log](https://github.com/pydata/numexpr/files/3996174/numexpr.log)

**Comment by daniel-klein** [comment]
While my previous attempt was using pip, I've since cloned the repo and attempted to build via `python setup.py build` according to the installation document.  Unfortunately, this also fails (log attached).

When I try to run the build command in Terminal, I get the following message:

```console
(IDM) [dklein@declinebook numexpr ]$ g++ -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -arch x86_64 -g -std=c99 -I/Users/dklein/venv/IDM/lib/python3.8/site-packages/numpy/core/include -I/Users/dklein/venv/IDM/include -I/Library/Frameworks/Python.framework/Versions/3.8/include/python3.8 -c numexpr/interpreter.cpp -o build/temp.macosx-10.9-x86_64-3.8/numexpr/interpreter.o -MMD -MF build/temp.macosx-10.9-x86_64-3.8/numexpr/interpreter.o.d -funroll-all-loops
clang: warning: optimization flag '-funroll-all-loops' is not supported [-Wignored-optimization-argument]
error: invalid argument '-std=c99' not allowed with 'C++'
```

I will next try downgrading my environment to python 3.7.

[build.log](https://github.com/pydata/numexpr/files/3996238/build.log)

**Comment by robbmcleod** [comment]
I'm under the impression that the official CPython compiler on OSX is `clang++` and not `g++`, so perhaps you have some environment variable (i.e. `CC` and/or `CXX`) that causes Python to use `g++` instead? Difficult thing for me to test, however, given the lack of options for OSX virtualization. Or I'm wondering if `g++` is aliased to something completely different on your system?

**Comment by daniel-klein** [comment]
I reverted to python 3.7.6 and it works like a charm.  So there must be some issue installing with the latest python (3.8.1) on macOS.

Thanks for the reply @robbmcleod.  For what it's worth, I haven't intentionally done anything super fancy with env variables or links.  Here's some info on my g++ version.

```command
(IDM) [dklein@declinebook URHI ]$ g++ --version
Configured with: --prefix=/Library/Developer/CommandLineTools/usr --with-gxx-include-dir=/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/4.2.1
Apple clang version 11.0.0 (clang-1100.0.33.8)
Target: x86_64-apple-darwin19.0.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
```

Happy to try a few things if it'd help with debugging.

**Comment by robbmcleod** [comment]
We have wheels available for OSX on Python 3.8 now for NumExpr 2.7.0, so you should be able to install via `pip` with 3.8 if you prefer. I've seen people with issues similar to yours in the past, it's generally some obscure configuration issue. I'll close the issue for now, since Travis CI is successfully building our Mac OSX wheels.

**Comment by daniel-klein** [comment]
Yup, with the latest wheels, it works great on 3.8.1.  Thanks!
